github/MainClass.py

```diff
@@ -118,12 +126,40 @@ def __init__(
             or isinstance(retry, urllib3.util.Retry)
         ), retry
         assert pool_size is None or isinstance(pool_size, int), pool_size
+        assert auth is None or isinstance(auth, Auth.Auth), auth
+
+        if password is not None:
+            warnings.warn(
+                "Arguments login_or_token and password are deprecated, please use "
+                "auth=github.Auth.Login(...) instead",
+                category=DeprecationWarning,
+            )
+            auth = Auth.Login(login_or_token, password)
+        elif login_or_token is not None:
+            warnings.warn(
+                "Argument login_or_token is deprecated, please use "
+                "auth=github.Auth.Token(...) instead",
+                category=DeprecationWarning,
+            )
+            auth = Auth.Token(login_or_token)
+        elif jwt is not None:
+            warnings.warn(
+                "Argument jwt is deprecated, please use "
+                "auth=github.Auth.AppAuth(...) or "
+                "auth=github.Auth.AppAuthToken(...) instead",
+                category=DeprecationWarning,
+            )
+            auth = Auth.AppAuthToken(jwt)
+        elif app_auth is not None:
+            warnings.warn(
+                "Argument app_auth is deprecated, please use "
+                "auth=github.Auth.AppInstallationAuth(...) instead",
+                category=DeprecationWarning,
+            )
+            auth = app_auth
```

Comment:
Well, I think that should be removed in the next major release (v2) as this is quite fundamental / widespread use.

We have seen lots of small breaking changes in minor releases, recently, and given there is a lot of refactoring still need, I would refrain from following semantic versioning.

Reply:
> What if no auth is provided, is that anonymous auth at that point? Should an `Auth.Auth` type for anonymous access be added too, just so that there isn't a 'None' case?

Good point. I think `auth=None` is expressive enough. I don't see benefit for an `NoAuth` or `Anonymous` auth class.